crates/scheduler/src/call/led.rs

```diff
@@ -44,10 +42,8 @@ fn count<B: Board>(call: SchedulerCall<B, api::count::Sig>) {
 fn get<B: Board>(call: SchedulerCall<B, api::get::Sig>) {
     let api::get::Params { led } = call.read();
     let result = try {
-        match Id::new(*led as usize) {
-            Some(id) => board::Led::<B>::get(id),
-            None => Err(Error::user(Code::InvalidArgument)),
-        }
+        let id = Id::new(*led as usize)?;
```

Comment:
This is incorrect for a very subtle reason. In `scheduler/src/lib.rs` there is an implementation of `From<Error> for Trap`. So here, we still trap instead of returning an error.

This should be:

```rust
let result = try {
    let id = Id::new(*led as usize)?;
    board::Led::<B>::get(id)?
};
call.reply(Ok(result));
```

Notice the `Ok(result)` at the end.

I think the next PR should be to remove that `impl From<Error> for Trap` and fix the compilation errors by manually converting to `Trap` with `.map_err(|_| Trap)`.

Reply:
Done. Thanks for pointing that out!